### ISSUE TITLE
HTML syntax highlighting is incorrect

### ISSUE THREAD

**Issue by Bluebie** [microsoft/vscode#140360]
Issue Type: <b>Bug</b>

if you enter in:

"<a href=https://example.org/>foo</a>" in a html highlighted document, the final / in the href attribute value is highlighted in a grey colour, as if <a> is a self closing xml tag. I believe this behaviour is deceptive, it gives the wrong impression that a html parser would understand the code in this way, but the parsed value of the tag's href attribute is indeed "https://example.org/", which is true in Chromium and in the HTML5 spec. 

VS Code version: Code 1.63.2 (Universal) (899d46d82c4c95423fb7e10e68eba52050e30ba3, 2021-12-15T09:37:28.172Z)
OS version: Darwin arm64 21.2.0
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 24)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|5, 4, 3|
|Memory (System)|8.00GB (0.11GB free)|
|Process Argv|--crash-reporter-id 853ff5a3-ca88-48f7-846c-f5ed1cf44a91|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (18)</summary>

Extension|Author (truncated)|Version
---|---|---
cddl-languageserver|anw|0.1.3
better-toml|bun|0.3.2
vscode-standardjs-snippets|cap|0.8.15
vscode-standardjs|che|1.4.1
vscode-eslint|dba|2.2.2
vs-code-xml-format|fab|0.1.5
auto-dark-mode|Lin|0.1.7
vscode-docker|ms-|1.18.0
remote-ssh|ms-|0.70.0
remote-ssh-edit|ms-|0.70.0
cpptools|ms-|1.7.1
nodejs-docs|Muk|0.2.2
vscode-hexdump|sle|1.8.1
es6-mocha-snippets|spo|0.2.2
svelte-vscode|sve|105.8.0
vscodeintellicode|Vis|1.2.15
five-server|yan|0.1.2
city-lights-icon-vsc|Yum|1.1.3

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
pythontb:30283811
pythonvspyt551cf:30345471
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscorecescf:30384386
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
vscop804:30404766
vscop341:30404997
vsrem710:30416614
vscexrecpromp3t1:30407762

```

</details>

**Comment by gjsjohnmurray** [comment]
The attribute value needs to be enclosed in quotes.

**Comment by Bluebie** [comment]
> The attribute value needs to be enclosed in quotes.

No, they don't. Quotes around attribute values that do not contain whitespace or `=`, `>`, `"`, `'`, <code>`</code> are redundant when it comes to parsing html documents.

https://html.spec.whatwg.org/multipage/parsing.html#attribute-value-(unquoted)-state

It should not be the responsibility of the syntax highlighter to impose opinion on what constitutes good or bad HTML. That should be the responsibility of linters.

**Comment by alexr00** [comment]
Thank you for the issue and the spec reference! I have opened an issue in the upstream repository that we get our html syntax highlighting grammar from: https://github.com/textmate/html.tmbundle/issues/118
